### PR TITLE
added initial game redux store

### PR DESCRIPTION
**Pull request by srashed001**
initial game reducer
- check changes in GameCard and GameDetails

### CODE REVIEW

File: src/store/games/gamesReducer.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+
+const INITIAL_STATE = {
```
Comment (swiftpaca) [inline src/store/games/gamesReducer.js]: Do you still need this file if you're creating the reducer in gamesSlice.js?

File: src/games/GameDetails.js [hunk 2]
```diff
@@ -1,53 +1,52 @@
...
-      setComments(comments)
-      setPlayers(players)
-    }
+  const { gameId } = useParams();
 
-    getGame()
-  }, [gameId])
+  const dispatch = useDispatch();
+  const gameStatus = useSelector((state) => state.games.status);
+  const games = useSelector((state) => state.games.entities);
```
Comment (swiftpaca) [inline src/games/GameDetails.js]: you can use this selector to get the exact game you need

File: src/games/GameDetails.js [hunk 3]
```diff
@@ -1,53 +1,52 @@
...
-        <li>state: {details.state}</li>
-        <li>created on: {details.createdOn}</li>
-        <li>game host: {details.createdBy}</li>
-        <li>days until game: {details.daysDiff}</li>
-      </ul>
-    </div>
-  );
+  useEffect(() => {
+    if (gameStatus === "idle") {
+      dispatch(fetchGames());
```
Comment (swiftpaca) [inline src/games/GameDetails.js]: Not sure how the Game Details is intended but if you're on the route `/games/$id` i dont think you'll end up fetching all the games on this page

**Comment by swiftpaca** [comment]
Overall looks like a solid start. Just a few small comments

File: src/store/games/gamesReducer.js (hunk 1 above)
Comment (srashed001) [reply src/store/games/gamesReducer.js]: file removed

File: src/games/GameDetails.js (hunk 2 above)
Comment (srashed001) [reply src/games/GameDetails.js]: updated

File: src/games/GameDetails.js (hunk 3 above)
Comment (srashed001) [reply src/games/GameDetails.js]: updated to fetchGame(gameId) 
fetchGame retrieves game data and hydrates gameSlice, userSlice, and commentSlice

File: src/api/api.js [hunk 4]
```diff
@@ -184,8 +184,8 @@ class PugApi {
   /** get invites for user */
 
   static async getInvites(username) {
-    let res = await this.request(`users/${username}/invites}`);
-    return res.invtes;
+    let res = await this.request(`users/${username}/invites`);
```
Comment (swiftpaca) [inline src/api/api.js:187]: this is a `const`

Comment (swiftpaca) [reply src/api/api.js:187]: actually it looks like most of these `res` variables are const. Not a big deal just being nitpicky

File: src/games/GamesList.js [hunk 5]
```diff
@@ -1,52 +1,51 @@
...
-    const [games, setGames] = useState(null)
+  const gameStatus = useSelector((state) => state.games.status);
+  const error = useSelector((state) => state.games.error);
 
-    async function search(data = {}){
-        const games = await PugApi.getGames(data);
-        setGames(games)
+  useEffect(() => {
+    console.log(`GamesList useEffect`);
+    if (gameStatus === "idle") {
```
Comment (swiftpaca) [inline src/games/GamesList.js:19]: Are you only checking for `fetched` state in the Details component?

File: src/users/UserDetails.js [hunk 6]
```diff
@@ -34,13 +58,39 @@ function UserDetails(){
...
-          <li>following: {bio.following}</li>
-          <li>followed: {bio.followed}</li>
+          <li>following: {bio.following.length}</li>
+          <li>followed: {bio.followed.length}</li>
           <li>is private: {JSON.stringify(bio.isPrivate)}</li>
           <li>is admin: {JSON.stringify(bio.isAdmin)}</li>
-        </ul>
+           </ul>
+          <ul>
+            games hosted pending:{" "}
```
Comment (swiftpaca) [inline src/users/UserDetails.js:67]: whats the intention here?

File: src/users/UsersList.js [hunk 7]
```diff
@@ -1,42 +1,51 @@
...
+    return () => {
+     dispatch(resetUserStatus())
+      console.log(userStatus)
+    }
   }, []);
 
-  if (!users) return <LoadingSpinner />;
+  useEffect(() => {
+    console.log("UserList useEffect");
+    if(userStatus === 'idle'){
```
Comment (swiftpaca) [inline src/users/UsersList.js:29]: Same question, are we only using the local `fetched` state for Details?

**Pull request merged**